File: backend/app/schemas/dbn_standard.py
"""DBN Standard schemas."""

from uuid import UUID

from pydantic import BaseModel


class CriterionCreate(BaseModel):
    name: str
    description: str | None = None
    weight: float
    max_score: float
    sort_order: int = 0


class CriterionResponse(BaseModel):
    id: UUID
    name: str
    description: str | None
    weight: float
    max_score: float
    sort_order: int

    model_config = {"from_attributes": True}


class StandardCreate(BaseModel):
    name: str
    description: str | None = None
    version: str


class StandardResponse(BaseModel):
    id: UUID
    name: str
    description: str | None
    version: str
    is_active: bool
    criteria: list[CriterionResponse] = []

    model_config = {"from_attributes": True}

File: frontend/src/types/dbn-standard.ts
export interface CriterionResponse {
  id: string;
  name: string;
  description: string | null;
  weight: number;
  max_score: number;
  sort_order: number;
}

export interface StandardResponse {
  id: string;
  name: string;
  description: string | null;
  version: string;
  is_active: boolean;
  criteria: CriterionResponse[];
}
